Add subscription api documentation (#127)
Yoast/developer-docs: docs/customization/myyoast/apis/subscription-api.md

---
id: subscription-api
title: MyYoast - Subscription API
sidebar_label: Subscription API

> description: For yoast partners to create and manage subscriptions.

Reviewer: ```suggestion
description: For Yoast partners to create and manage subscriptions.
```

<Alert>

If you wish to become a Yoast partner, please contact [Joost](https://yoast.com/about-us/team/joost-de-valk/) or [Thijs](https://yoast.com/about-us/team/thijs-de-valk/).
</Alert>

We have a Swagger document you can use to [explore the API](https://my.yoast.com/provisioning-api/#/). We can provide the credentials to view it.

## Provisioner

A provisioner is a Yoast partner allowed to create subscriptions in MyYoast.

We create an account for each provisioner, and with the credentials for that account, the provisioners can call the API endpoints.

The API uses basic auth. If you are a provisioner and do not yet have the proper credentials to call the provisioning API, please contact us.

## Client API

To assist provisioners we have used [swagger-codegen](https://swagger.io/docs/open-source-tools/swagger-codegen/) to generate clients for the API. There's:

- [PHP client](https://github.com/Yoast/subscription-api-php-client)
    - [With additional documentation](https://github.com/Yoast/subscription-api-php-client/blob/master/docs/Api/SubscriptionProvisioningApi.md)
- [javascript client](https://github.com/Yoast/subscription-api-javascript-client)

> The readme of these clients should explain how to install and use them.

Reviewer: ```suggestion
The readme of these clients explains how to install and use them.
```

Renewing a subscription sets the end date of the subscription to a new date further in the future.
If the end date of a subscription is in the past, the amount of extra time is added to the date of the request instead.

**Cancelled subscriptions can no longer be renewed.**

#### Cancelling subscriptions

If it is known that a customer will not renew their subscription or if the customer should no longer have their subscription, the [cancel endpoint](https://my.yoast.com/provisioning-api/#/Subscription%20Provisioning/post_api_provisioning_subscriptions__id__cancel) should be used.

The status of that subscription will be set to `pending-cancel`. It's still a valid license until the end date passes, then it will be set to `cancelled`.

If the customer should lose access immediately. you can set `immediately` to `true` in the body. In this case, the subscription is set to `cancelled` immediately, regardless of the amount of time the subscription had remaining. The subscription is no longer considered a valid license when it has the `cancelled` status.

### Subscription expiration and cancellation.

If a subscription's end date has passed, that subscription is due for expiration.

> If the subscription's status is `active`, the subscription's status will be set to `expired`, and the subscription is no longer considered a valid license. We have a CRON job in my-yoast responsible for this. This is done automatically every 30 minutes.

Reviewer: ```suggestion
If the subscription's status is `active`, the subscription's status will be set to `expired`, and the subscription is no longer considered a valid license. This is done automatically every 30 minutes.
```
